OTRExporter/OTRExporter/Main.cpp

```diff
@@ -67,6 +68,47 @@ static void ExporterProgramEnd()
 {
 	uint32_t crc = 0xFFFFFFFF;
 	const uint8_t endianness = (uint8_t)Endianness::Big;
+
+	std::vector<int16_t> sohVersion = {};
+	std::vector<std::string> versionParts = StringHelper::Split(sohVersionString, ".");
+
+	for (auto val : versionParts)
+	{
+		int16_t num = 0;
+		try
+		{
+			num = std::stoi(val, nullptr);
+		}
+		catch (std::invalid_argument &e)
+		{
+			num = 0;
+		}
+		catch (std::out_of_range &e)
+		{
+			num = 0;
+		}
+
+		sohVersion.push_back(num);
+
+		// Ignore any extra values passed in
+		if (sohVersion.size() == 3)
+			break;
+	}
+
+	// Padded out missing values
+	while (sohVersion.size() < 3)
+	{
+		sohVersion.push_back(0);
+	}
```

Comment:
i guess i was just thinking
```suggestion
	// Padded out missing values
	if (sohVersion.size() != 3)
	{
		sohVersion = { 0, 0, 0 };
	}
```
so that way we either get a correctly parsed value or zeros

Reply:
I've simplified the version parsing since yeah only SoH will ever execute this internally, so it should ideally never fail.